bokeh/sphinxext/bokehjs_link.py

```diff
@@ -0,0 +1,108 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2020, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+''' Simplify linking to versioned BokehJS CDN urls.
+
+This module provides a Sphinx role that can be used to easily link to various
+BokehJS releases:
+
+``:bokehjs-link:`` : BokehJS file
+
+Examples
+--------
+
+The following code:
+
+.. code-block:: rest
+
+    The current version of BokehJS widgets and tables can be
+    downloaded at:
+    * :bokehjs-link:`bokeh-widgets`
+    * :bokehjs-link:`bokeh-tables`
+
+yields the output:
+
+The current version of BokehJS widgets and tables can be
+downloaded at:
+* :bokehjs-link:`bokeh-widgets`
+* :bokehjs-link:`bokeh-tables`
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# External imports
+from docutils import nodes
+from docutils.parsers.rst.roles import set_classes
+
+# Bokeh imports
+from .util import get_sphinx_resources
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'bokehjs_link',
+    'setup',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Dev API
+#-----------------------------------------------------------------------------
+
+
+def bokehjs_link(name, rawtext, text, lineno, inliner, options=None, content=None):
+    ''' Download link to versioned BokehJS CDN url.
+
+    Returns 2 part tuple containing list of nodes to insert into the
+    document and a list of system messages.  Both are allowed to be
+    empty.
+
+    '''
+    resources = get_sphinx_resources(include_bokehjs_api=True)
+
+    js_files = resources.js_files
+
+    if text == 'bokeh':
+        url = js_files[0]
+    elif text == 'bokeh-widgets':
+        url = js_files[1]
+    elif text == 'bokeh-tables':
+        url = js_files[2]
+    elif text == 'bokeh-gl':
+        url = js_files[3]
+    elif text == 'bokeh-api':
+        url = js_files[4]
```

Comment:
FYI this order is not guaranteed to be stable and if new bundles are added it is guaranteed to change. I think it would be safer to loop and pattern match until the right one is found.

Reply:
Yea I wasn't too pleased with this logic. What about adding a new method in the Resources class that returns a dictionary where the key is the component and the value is the url? Like so:
```python
urls = {
    'bokeh-widgets': 'https://cdn.bokeh.org/bokeh/release/bokeh-widgets-2.0.1.min.js',
    'bokeh-tables': 'https://cdn.bokeh.org/bokeh/release/bokeh-tables-2.0.1.min.js',
    'bokeh-api': 'https://cdn.bokeh.org/bokeh/release/bokeh-api-2.0.1.min.js',
}
```
In any case, this should be improved so I can start with the pattern matching.

Reply:
Pattern matching works:
<img width="1395" alt="Screen Shot 2020-05-29 at 6 10 10 PM" src="https://user-images.githubusercontent.com/7004949/83315890-b4b4a680-a1d7-11ea-90e9-9321b6775c43.png">

Reply:
Works for dev builds too:
<img width="1395" alt="Screen Shot 2020-05-30 at 1 21 09 PM" src="https://user-images.githubusercontent.com/7004949/83338461-7cfe3b00-a279-11ea-8fe4-abe50c99ddd7.png">